examples/grid.cpp

```diff
@@ -305,42 +397,94 @@ DrmRenderer::Flipper::~Flipper()
 
 bool DrmRenderer::Flipper::init()
 {
+    if (!m_fpsPlane->init())
+        return false;
     if (pthread_create(&m_thread, NULL, start, this)) {
         ERROR("create thread failed");
         return false;
     }
     return true;
 }
 
+bool DrmRenderer::Flipper::waitingRenderTime()
+{
+    bool late = false;
+    if (!m_fps)
+        return late;
+    timeval current;
+    if (m_firstFrame) {
+        m_firstFrame = false;
+        gettimeofday(&m_nextTime, NULL);
+        m_duration.tv_sec = 0;
+        m_duration.tv_usec = 1000 * 1000 / m_fps;
+    }
+    else {
+        bool neverSleep = true;
+        do {
+            gettimeofday(&current, NULL);
+            timeval refresh;
+            refresh.tv_sec = 0;
+            refresh.tv_usec = 1000 * 1000 / 60;
+            timeval nextRefresh;
+            timeradd(&current, &refresh, &nextRefresh);
+
+            if (timercmp(&nextRefresh, &m_nextTime, < )) {
+                usleep(refresh.tv_usec);
+                neverSleep = false;
+            }
+            else {
+                if (neverSleep) {
+                    late = !timercmp(&current, &m_nextTime, < );
+                    /*       if (late) {
```

Comment:
Better not. This will help debug the performance, but we do not want it printout or even execute in normal case. Therefore, I comment out them here.